src/main/java/com/bwell/auth/service/AuthService.java

```diff
@@ -0,0 +1,76 @@
+package com.bwell.auth.service;
+
+import com.bwell.auth.model.Client;
+import com.bwell.auth.model.Token;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections4.map.PassiveExpiringMap;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Base64;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+
+public class AuthService {
+    private static final String SCOPE = "scope";
+    private static final String GRANT_TYPE = "grant_type";
+    private static final String CLIENT_CREDENTIALS = "client_credentials";
+
+    private static final RestTemplate restTemplate;
+    private static final ObjectMapper objectMapper;
+    private static final PassiveExpiringMap<String, Token> cache;
+
+    static {
+        objectMapper = new ObjectMapper();
+        objectMapper.writerWithDefaultPrettyPrinter();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        cache = new PassiveExpiringMap<>(60, TimeUnit.MINUTES);
+
+        restTemplate = new RestTemplate();
+    }
+
+    public String token(Client client) throws JsonProcessingException {
```

Comment:
I can rename it. 

The logic to check if its expired is in the PassiveExpiringMap. The Map is hardcoded at 60 minutes to remove the Token from the Map. In the token method, if the map is empty, it will go and get a new token. So, the Map is kicking it out. 

It's hard coded to 60 mins for now.

Reply:
Method is renamed